src/param.jl

```diff
@@ -79,13 +80,13 @@ end
 Param(val; kwargs...) = Param((; val=val, kwargs...))
 Param(; kwargs...) = Param((; kwargs...))
 
+setparent(::P, newparent) where P<:AbstractParam = P.name.wrapper(newparent)
```

Comment:
```suggestion
setparent(::P, newparent) where P<:AbstractParam = ConstructionBase.constructorof(P)(newparent)
```

Something like this would be more generic.